packages/hoprd/src/index.ts

```diff
@@ -246,13 +31,214 @@ function addUnhandledPromiseRejectionHandler() {
   })
 }
 
-async function main() {
+export async function main() {
   // Starting with Node.js 15, undhandled promise rejections terminate the
   // process with a non-zero exit code, which makes debugging quite difficult.
   // Therefore adding a promise rejection handler to make sure that the origin of
   // the rejected promise can be detected.
   addUnhandledPromiseRejectionHandler()
 
+  const argv = yargs(process.argv.slice(2))
+    .option('environment', {
+      string: true,
+      describe: 'Environment id which the node shall run on',
+      choices: supportedEnvironments().map((env) => env.id),
+      default: DEFAULT_ENVIRONMENT
```

Comment:
This is incorrect imho. Now releases have a default environment which doesn't make sense. The idea of using the helper function was to have release-specific default environments set instead of one generic hardhat default.

Reply:
Discussed internally, this change sadly overlooks the fact that for our docker builds we _do_ use the `default-environment.json` file which "bakes" our images to a given release. Ideally, we should do this on [runtime](https://12factor.net/build-release-run) but this works due to our current limitations, and that's what we had been documenting so far.

Additionally, this was initially to support hoprnet/hopr-devrel#168 which we will soon close for reasons that will be documented in the issue. Finally, this introduced a bug patched in #3385 which now we can close if reverting. Thus, we'll be reverting these changes.